openpype/hosts/maya/plugins/publish/extract_xgen.py

```diff
@@ -0,0 +1,139 @@
+import os
+import copy
+
+from maya import cmds
+import pymel.core as pc
+import xgenm
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api.lib import maintained_selection, attribute_values
+from openpype.lib import StringTemplate
+
+
+class ExtractXgenCache(publish.Extractor):
+    """Extract Xgen"""
+
+    label = "Extract Xgen"
+    hosts = ["maya"]
+    families = ["xgen"]
+    scene_type = "ma"
+
+    def process(self, instance):
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        staging_dir = self.staging_dir(instance)
+        maya_filename = "{}.{}".format(instance.data["name"], self.scene_type)
+        maya_filepath = os.path.join(staging_dir, maya_filename)
+
+        # Get published xgen file name.
+        template_data = copy.deepcopy(instance.data["anatomyData"])
+        template_data.update({"ext": "xgen"})
+        templates = instance.context.data["anatomy"].templates["publish"]
+        xgen_filename = StringTemplate(templates["file"]).format(template_data)
+        name = instance.data["xgenPalette"].replace(":", "__").replace("|", "")
+        xgen_filename = xgen_filename.replace(".xgen", "__" + name + ".xgen")
+
+        # Export xgen palette files.
+        xgen_path = os.path.join(staging_dir, xgen_filename).replace("\\", "/")
+        xgenm.exportPalette(
+            instance.data["xgenPalette"].replace("|", ""), xgen_path
+        )
+        self.log.info("Extracted to {}".format(xgen_path))
+
+        representation = {
+            "name": name,
+            "ext": "xgen",
+            "files": xgen_filename,
+            "stagingDir": staging_dir,
+        }
+        instance.data["representations"].append(representation)
+
+        # Collect nodes to export.
+        duplicate_nodes = []
+        for node, connections in instance.data["xgenConnections"].items():
+            transform_name = connections["transform"].split(".")[0]
+
+            # Duplicate_transform subd patch geometry.
+            duplicate_transform = pc.duplicate(transform_name)[0]
+            pc.parent(duplicate_transform, world=True)
+            duplicate_transform.name(stripNamespace=True)
```

Comment:
This query is unused. `.name()` as far as I know returns a value and doesn't change the actual node. So `stripNamespace=True` here does nothing to the node it's acting upon.

Reply:
Good point.